habitTrackerApp.py

```diff
@@ -97,7 +96,7 @@ def callback(instance, value):
         while date_cursor.month == self.date.month:
             date_label = OtherButton(group = 'cat')
             date_label.otherButtonID = date_cursor
-            date_label.text = str(date_label.otherButtonID)
+            date_label.text = str(date_cursor.day)
             date_label.bind(on_press=partial(self.set_date, day=date_cursor.day))
             date_label.bind(state=callback)
             if date_cursor == date.today():
```

Comment:
the inside of this "if" (line 103 to 107) could be in a function, this help understanding what's happening. 
I see that you change the date_label  and the color etc, I am guessing that you are highlighting the date. 
Encapsulating this into a function called might "highlightDate()" (or whatever it is that you are doing) convey what you want to do and make the code easier to read

note this advice applies to several place in the code